apps/emqx_dashboard/src/emqx_dashboard_admin.erl

```diff
@@ -91,23 +93,21 @@ add_user_(Admin = #?ADMIN{username = Username}) ->
 remove_user(Username) when is_binary(Username) ->
     Trans = fun() ->
                     case lookup_user(Username) of
-                    [] ->
-                        mnesia:abort(<<"Username Not Found">>);
-                    _  -> ok
-                    end,
-                    mnesia:delete({?ADMIN, Username})
+                        [] -> mnesia:abort(<<"Username Not Found">>);
+                        _  -> mnesia:delete({?ADMIN, Username})
+                    end
             end,
     return(mria:transaction(?DASHBOARD_SHARD, Trans)).
 
 -spec(update_user(binary(), binary()) -> ok | {error, term()}).
-update_user(Username, Tags) when is_binary(Username) ->
-    return(mria:transaction(?DASHBOARD_SHARD, fun update_user_/2, [Username, Tags])).
+update_user(Username, Desc) when is_binary(Username) ->
+    return(mria:transaction(?DASHBOARD_SHARD, fun update_user_/2, [Username, Desc])).
 
 %% @private
-update_user_(Username, Tags) ->
+update_user_(Username, Desc) ->
     case mnesia:wread({?ADMIN, Username}) of
         [] -> mnesia:abort(<<"Username Not Found">>);
-        [Admin] -> mnesia:write(Admin#?ADMIN{tags = Tags})
+        [Admin] -> mnesia:write(Admin#?ADMIN{description = Desc})
```

Comment:
Return `NewAdmin` instead of 'ok'.
the change the return/1 to 
```erlang
return({atomic, Result}) ->
    Result;
```

then we don't need search user after update

Reply:
the result returned by api should be a map.
but the `Result` in `{atomic, Result}` is the value of the last expression in
`Fun` (applied by `mria:transaction(?TAB, Fun, [Args]` ).
it returned `'ok'` by `mnesia:write/1`